tests/unit/package_managers/yarn_classic/test_utils.py

```diff
@@ -0,0 +1,69 @@
+from unittest import mock
+
+from cachi2.core.package_managers.yarn_classic.project import PackageJson
+from cachi2.core.package_managers.yarn_classic.utils import find_runtime_deps
+from cachi2.core.rooted_path import RootedPath
+
+PACKAGE_JSON_CONTENT = """
+{
+  "name": "project",
+  "dependencies": {
+    "main-dep1": "^1.0.0"
+  },
+  "optionalDependencies": {
+    "optional-dep1": "^2.0.0"
+  },
+  "peerDependencies": {
+    "peer-dep1": "^3.0.0"
+  },
+  "devDependencies": {
+    "dev-dep1": "^4.0.0"
+  }
+}
+"""
+
+
+@mock.patch("cachi2.core.package_managers.yarn_classic.project.YarnLock")
+@mock.patch("cachi2.core.package_managers.yarn_classic.utils._expand_yarn_lock_keys")
+def test_find_runtime_deps(
+    mock_expand_yarn_lock_keys: mock.Mock,
+    mock_yarn_lock: mock.Mock,
+    rooted_tmp_path: RootedPath,
+) -> None:
+    rooted_tmp_path.join_within_root("package.json").path.write_text(PACKAGE_JSON_CONTENT)
+    package_json = PackageJson.from_file(rooted_tmp_path.join_within_root("package.json"))
+
+    mock_expand_yarn_lock_keys.return_value = {
+        # dependencies
+        "main-dep1@^1.0.0": {
+            "version": "1.0.0",
+            "dependencies": {"sub-dep1": "^1.1.0"},
+        },
+        # optional dependencies
+        "optional-dep1@^2.0.0": {
+            "version": "2.0.0",
+            "dependencies": {"multi-dep1": "^4.0.0", "multi-dep2": "^4.0.0"},
+        },
+        "multi-dep1@^4.0.0, multi-dep2@^4.0.0": {"version": "5.0.0", "dependencies": {}},
```

Comment:
Isn't `_expand_yarn_lock_keys` supposed to split this key in two?

Reply:
yes

```
  assert result == {
      ...
      "multi-dep1@^4.0.0",
      "multi-dep2@^4.0.0",
 ```

Reply:
Sigh, the mock of `_expand_yarn_lock_keys()` shouldn't definitely be there :disappointed: Fixed.

Reply:
Thank you, now the test is much clearer! I do think that you would want to either rename or drop the compound key entry. I suggest renaming them to something like `compund-multi-dep1` to exercise the splitting path too.